scripts/scripts.js

```diff
@@ -120,6 +120,35 @@ export function formatDate(date) {
   });
 }
 
+/**
+ * Builds an article sidebar and appends it to main in a new section.
+ * @param main
+ */
+async function buildArticleSidebar(main) {
+  if (main.querySelector('.sidebar')) {
+    // already got a sidebar, !main.firstChild avoids recursive rendering for empty `main` blocks
+    return;
+  }
+  const locInfo = getLocaleInfo();
+  const sidebarBlock = buildBlock('sidebar', [
+    [a({ href: `${locInfo.placeholdersPrefix}/fragments/sidebar-fragment` }, 'Sidebar')],
+  ]);
+
+  const sidebar = div(sidebarBlock);
+  main.append(sidebar);
+
+  main.classList.add('has-sidebar');
+  const sidebarOffset = 2;
+
+  const numSections = main.children.length - 1;
+  main.style = `grid-template-rows: repeat(${numSections}, auto);`;
+
+  sidebar.style.gridRow = `${sidebarOffset} / infinite`;
+  for (let i = 0; i < sidebarOffset - 1; i += 1) {
+    main.children[i].classList.add('no-sidebar');
+  }
```

Comment:
Having the code to build the sidebar duplicated creates the potential for having it broken if somebody changes the logic in one place, but not the other. Could we find a better alternative to do this directly in the `detectSidebar` function?